### ISSUE TITLE
Design and implement a "show stats" action

### ISSUE THREAD

**Issue by jbrouwer5** [uchicago-cs/chiventure#1290]
The Graphics team is working on a stats menu that can display current character stats. Action Management needs to create an action `view` that allows the user to view these stats in a large pop up menu. We plan on implementing a new action type to cover this new action. Doing this will require a new function for handling these new actions. This new function will likely follow the model of previous action functions.

**Comment by KendrickXie** [comment]
Our plan for this issue is as follows:

1. Add new action kind in action_structs.h and c
2. Add `do_player_action` in actionmanagement.h and c in action potentially have a player as a parameter call display_stats from the game-state/stats.h module (this needs a pointer to the stats hash table to be printed)
3. Create an example to show the graphics team how to use the action

**Comment by HarrisonD123** [comment]
We determined that the `game-state/game-action.h` function has a checker function called `possible_action()` that only deals with items. It may be necessary in the future when we have more actions to abstract this function to handle any action.

We eventually decided that there was no need to implement a possible action function for showing stats given that any player should be allowed to see their stats.

We also determined that the `display_stats()` function in the `game-state/stats.h` module utilizes a hash iterator (essentially a for loop in functionality) from the uthash C library to print each stat in the format of:
`<Stat Name> [<Stat Value> / <Stat Max>]
`
We plan on calling this function in our `do_player_action` in order to return this same string.

**Comment by HarrisonD123** [comment]
I just implemented a new action type in the `action` enum and `action_kind` struct of the `action_structs.h` module to accommodate this issue. The new action is `VIEW` (for VIEW Stats) and the `action_kind` is `self`, which will allow for a whole set of potential player-related actions.

**Comment by HarrisonD123** [comment]
Added a new `self_action_object` enum to allow for multiple different objects about a player to be viewed. This enum currently contains `STATS,` but may eventually include things like `INVENTORY` and `MAP,` etc. I added this enum as a parameter in the `do_self_action` function and wrote out corresponding logic as an if statement.

**Comment by HarrisonD123** [comment]
**Closing Statement**

The following changes have been implemented:

- `actions` enum: New action VIEW added as an enum under a new action type we called "self"
- `action_kind` enum: New action type SELF added as our 4th action kind
- `self_action_object` enum: New enum created to accommodate a whole class of future display-centric actions. It currently only includes STATS, which is meant to be used when the user gives the `"view stats"` command
- `do_self_action` function: New function to handle actions that deal with player info. It can currently only handle the `"view stats"` command. Given this command, the function calls the `display_stats` function found in the `game-state/stats` module.

The `do_self_action` function currently only handles actions named `"view"` given the `STATS` enum. The logic is currently fairly primitive, implemented with a simple nested if statement, since we currently only have the case of `"view stats"`. There is potential make this implementation deeper with new structs and helper functions once we have implemented more self actions.

Jackson added the new enums we needed and came up with ideas for how ```do_self_action``` would work. Harrison and Kendrick worked on implementing ```do_self_action```.
